### DISCUSSION TITLE
What collectors used in RPI to send data to TSDBs in general?

### DISCUSSION THREAD

**Discussion by ylin30** [ytyou/ticktock#38] [General]
In desktop/servers, there are lots of data collectors frameworks (such as TCollector by Opentsdb, Telegraf by InfluxDB etc) to collect data and send to TSDBs. But these kind of collectors are heavyweight to RPIs. 

I am curious what kind of data collectors people normally used in RPIs? Do they just write their own Python/C program to collect data and send to TSDBs? 

I know MQTT is popular in RPIs. But you still need to write your own collectors and send them to MQTT. Why don't you send directly to TSDBs? 

Any inputs would be appreciated and helpful for us to serve RPIs better.

**Comment by juggledad** [comment]
I'm using Node-RED (work in progress) to send the data to ticktock. I have a mixture of WeMos D1 Mini's, Esp32 S2's and grabbing data from [OpenWeatherMap](http://openweathermap.org/appid). taking measurements, mostly Temperature, Humidity and Pressure. The devices send the data to NR via MQTT. With MQTT I can have each device setup a Last Will and Testament that will be send if the device goes off line. When NR receives the LTW it can be setup to send an email, Telegram msg or a Mastodon alert.

I've currently got ticktock and Node-RED running on a RPI B model 2. (I had one lying around)

**Reply by ylin30** [reply]
@juggledad Thanks for your input!

**Comment by juggledad** [comment]
I've posted thread on the Node-RED discourse forum with my instructions for installing ticktockdb and a flow for storing data and graphing it 
https://discourse.nodered.org/t/using-ticktockdb-an-open-source-time-series-database-on-a-rpi-2-model-b-and-node-red/84873

**Comment by ylin30** [comment]
@juggledad 👍👍👍
